packages/sit-onyx/src/components/OnyxToast/OnyxToast.vue

```diff
@@ -0,0 +1,253 @@
+<script lang="ts" setup>
+import circleAttention from "@sit-onyx/icons/circle-attention.svg?raw";
+import circleCheck from "@sit-onyx/icons/circle-check.svg?raw";
+import circleInformation from "@sit-onyx/icons/circle-information.svg?raw";
+import circleX from "@sit-onyx/icons/circle-x.svg?raw";
+import xSmall from "@sit-onyx/icons/x-small.svg?raw";
+import { computed } from "vue";
+import { useDensity } from "../../composables/density";
+import { injectI18n } from "../../i18n";
+import OnyxIcon from "../OnyxIcon/OnyxIcon.vue";
+import type { OnyxToastProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxToastProps>(), {
+  color: "neutral",
+  duration: 5000,
+  clickable: false,
+  icon: undefined, // needed to prevent default value being "false"
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the toast is clicked. Requires `clickable` property to be enabled.
+   */
+  click: [];
+  /**
+   * Emitted when the toast should be closed.
+   */
+  close: [];
+}>();
+
+const { t } = injectI18n();
+const { densityClass } = useDensity(props);
+
+const hasProgressBar = computed(() => props.duration > 0);
+
+const DEFAULT_ICONS: Record<typeof props.color, string> = {
+  neutral: circleInformation,
+  danger: circleX,
+  warning: circleAttention,
+  success: circleCheck,
+};
+
+const icon = computed(() => {
+  if (props.icon === false) return;
+  return props.icon || DEFAULT_ICONS[props.color];
+});
+</script>
+
+<template>
+  <component
+    :is="props.clickable ? 'button' : 'div'"
+    class="onyx-toast"
+    :class="[`onyx-toast--${props.color}`, densityClass]"
+    role="alert"
```

Comment:
Based on the color, we should switch the role: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions#roles_with_implicit_live_region_attributes

alert should only be used for important messages.

Reply:
I changed it to only have alert role for danger and warning color.
Should we use a special role for neutral and success colors?

Reply:
I would use `log` or `status` otherwise